crates/valence_core_macros/src/packet.rs

```diff
@@ -55,6 +85,34 @@ fn parse_packet_helper_attr(attrs: &[Attribute]) -> Result<Option<PacketAttr>> {
                 } else if meta.path.is_ident("tag") {
                     res.tag = Some(meta.value()?.parse::<LitInt>()?.base10_parse::<i32>()?);
                     Ok(())
+                } else if meta.path.is_ident("name") {
+                    res.name = Some(meta.value()?.parse::<LitStr>()?);
+                    Ok(())
+                } else if meta.path.is_ident("side") {
+                    let path = meta.value()?.parse::<Path>()?;
+                    let Some(first) = path.segments.first() else {
+                        return Err(meta.error("side path should have length equals 2"));
+                    };
+
+                    if first.ident != "PacketSide" {
+                        return Err(meta.error("side must starts with `PacketSide`"));
+                    }
+
+                    res.side = Some(path);
+
+                    Ok(())
+                } else if meta.path.is_ident("state") {
+                    let path = meta.value()?.parse::<Path>()?;
+                    let Some(first) = path.segments.first() else {
+                        return Err(meta.error("state path should have length equals 2"));
+                    };
+
+                    if first.ident != "PacketState" {
+                        return Err(meta.error("state must starts with `PacketState`"));
+                    }
+
+                    res.state = Some(path);
+                    Ok(())
```

Comment:
Might be better to treat these arguments as expressions rather than paths or literals for more flexibility.